server/controllers/macs.controller.js

```diff
@@ -26,82 +25,63 @@ macs.addMacs = function (req, res) {
         device: req.body.device
     };
 
-    MacModel.find({ 'mac': sample.mac }, fieldsIngored, function (err, data) {
-        if (data.length === 0) {
-            err = __addMac(sample);
-        } else if (data.length !== 0) {
-            err = __updateMac(sample, data);
-        }
-
-        if (err) {
-            res.status(500).send(err.message);
+    try {
+        if (Object.prototype.toString.call(_dateStringDate(req.body.origin.time)) !== '[object Date]') {
+            res.status(400).send('Invalid date');
+            logger.log('error', 'Invalid date');
+        } else if (typeof sample.mac !== 'string' || sample.mac.length !== 17 || sample.mac.split(':').length !== 6) {
```

Comment:
We will probably be storing hashes soon, for privacy and protection of personal data purposes, so this may have to be redone.